src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests_ListPatterns.cs

```diff
@@ -2699,6 +2699,60 @@ void verify(VarPatternSyntax declaration, string name, string expectedType)
         }
     }
 
+    [Fact]
+    public void SlicePattern_Nullability_NullableValue()
+    {
+        var source = @"
+#nullable enable
+class C
+{
+    public int Length => throw null!;
+    public int this[int i] => throw null!;
+    public int[]? Slice(int i, int j) => throw null!;
+
+    public void M()
+    {
+        if (this is [1, ..var slice])
+            slice.ToString();
+        if (this is [1, ..[] list])
+            list.ToString();
+    }
```

Comment:
Both warn about dereferenced of a possible null because we never check the value for null. 
This type is malformed under our well-behaved assumption, so the output is nonsensical.